owncloudApp/src/main/java/com/owncloud/android/workers/AutomaticUploadsWorker.kt

```diff
@@ -248,10 +247,13 @@ class AutomaticUploadsWorker(
         val arrayOfLocalFiles = documentTree?.listFiles() ?: arrayOf()
 
         val filteredList: List<DocumentFile> = arrayOfLocalFiles
+            .asSequence()
+            .filter {
+                it.lastModified() in lastSyncTimestamp..<currentTimestamp &&
```

Comment:
I wonder if it would make more sense to move the `MimetypeIconUtil` check before the `lastModified` check. The rationale being that the mime type lookup is hopefully just a hash table lookup whereas the modification check has to touch the filesystem. This would save quite a bit of I/O for cases where not many files match (let's say automatic video upload and the folder containing photos & videos consists of only 1% of videos, in such a case you'd have 99% useless I/O calls -- I actually think that such a case is probably not uncommon, even if one has a 50:50 ratio of pictures vs photos you'd check half the file unnecessarily).